### PR TITLE
Add journald support to Promtail amd64 binary builds

### PR DESCRIPTION
**Pull request by JordanRushing**
**What this PR does / why we need it**:

Previously, in https://github.com/grafana/loki/pull/8131, we enabled journald support for Promtail arm/arm64 binary builds but not amd64.

**Which issue(s) this PR fixes**:

- https://github.com/grafana/loki/issues/9060

**Special notes for your reviewer**:

N/A

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: .drone/drone.yml [hunk 1]
```diff
@@ -1463,6 +1463,7 @@ steps:
     NFPM_PASSPHRASE:
       from_secret: gpg_passphrase
     NFPM_SIGNING_KEY_FILE: /drone/src/private-key.key
+    PROMTAIL_JOURNAL_ENABLED: "true"
```
Comment (DylanGuedes) [inline .drone/drone.yml:1466]: could you confirm whether "true" here should be in string or boolean ("true" vs true like at line 1454)?

**Review by DylanGuedes — CHANGES_REQUESTED** [review]

File: .drone/drone.yml (hunk 1 above)
Comment (JordanRushing) [reply .drone/drone.yml:1466]: Good call; let me check

**Review by jeschkies — CHANGES_REQUESTED** [review]
Please also change the `dist` target for ARM builds. We pass the promtail journald tag explicitly which we probably should not be doing.

File: .drone/drone.jsonnet [hunk 2]
```diff
@@ -837,6 +837,7 @@ local manifest_ecr(apps, archs) = pipeline('manifest-ecr') {
             GITHUB_TOKEN: { from_secret: github_secret.name },
             NFPM_PASSPHRASE: { from_secret: gpg_passphrase.name },
             NFPM_SIGNING_KEY_FILE: '/drone/src/private-key.key',
+            PROMTAIL_JOURNAL_ENABLED: 'true',
```
Comment (jeschkies) [inline .drone/drone.jsonnet:840]: I think we should add these to `test packaging` as well to ensure that the build image has all dev dependencies required for building promtail with journald enabled.

**Review by jeschkies — APPROVED** [review]

**Review by trevorwhitney — APPROVED** [review]
lgtm

**Pull request merged**